src/core/SurfaceNode.cc

```diff
@@ -234,25 +235,25 @@ const Geometry *SurfaceNode::createGeometry() const
 
       double vx = (v1 + v2 + v3 + v4) / 4;
 
-      p->append_poly(3);
-      p->append_vertex(ox + j - 1, oy + i - 1, v1);
-      p->append_vertex(ox + j, oy + i - 1, v2);
-      p->append_vertex(ox + j - 0.5, oy + i - 0.5, vx);
-
-      p->append_poly(3);
-      p->append_vertex(ox + j, oy + i - 1, v2);
-      p->append_vertex(ox + j, oy + i, v4);
-      p->append_vertex(ox + j - 0.5, oy + i - 0.5, vx);
-
-      p->append_poly(3);
-      p->append_vertex(ox + j, oy + i, v4);
-      p->append_vertex(ox + j - 1, oy + i, v3);
-      p->append_vertex(ox + j - 0.5, oy + i - 0.5, vx);
-
-      p->append_poly(3);
-      p->append_vertex(ox + j - 1, oy + i, v3);
-      p->append_vertex(ox + j - 1, oy + i - 1, v1);
-      p->append_vertex(ox + j - 0.5, oy + i - 0.5, vx);
+      ind1=builder.vertexIndex(Vector3d(ox + j - 1, oy + i - 1, v1));
+      ind2=builder.vertexIndex(Vector3d(ox + j, oy + i - 1, v2));
+      ind3=builder.vertexIndex(Vector3d(ox + j - 0.5, oy + i - 0.5, vx));
+      builder.append_poly({ind1,ind2,ind3});
```

Comment:
This feels like a task for `PolySetBuilder::append_poly(const std::vector<Vector3d> &v)`